File: admin-be/src/main/java/com/bsep/admin/model/Alarm.java
package com.bsep.admin.model;

import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document("alarms")
public class Alarm {
    @Id
    private UUID id;
    private String name;
    private String text;
    private DeviceType deviceType;
    private LocalDateTime timestamp;
    private String deviceId;
}

File: admin-fe/src/app/model/alarm.ts
import { Log } from "./log";
import { Device, RealEstate } from "./myHouse";

export interface Alarm {
  id: string;
  name: string;
  text: string;
  deviceType?: string;
  timestamp: string;
  device: Device;
  realEstate: RealEstate;
}

export class LogAlarm {
  id: string = '';
  name: string = '';
  text: string = '';
  timestamp: Date | null = null;
  logs: Log[] = [];
}